src/components/visualizations/chart/chartOptionsBuilder.ts

```diff
@@ -1337,15 +1337,21 @@ function createYAxisItem(measuresInAxis: any[], opposite = false) {
 }
 
 function assignYAxes(series: any, yAxes: IAxis[]) {
-    return series.reduce((result: any, _: any, index: number) => {
+    return series.reduce((result: any, item: any, index: number) => {
         const yAxisIndex = yAxes.findIndex((axis: IAxis) => {
             return includes(get(axis, 'seriesIndices', []), index);
         });
         // for case viewBy and stackBy have one attribute, and one measure is sliced to multiple series
         // then 'yAxis' in other series should follow the first one
-        result[index].yAxis = yAxisIndex !== -1 ? yAxisIndex : result[0].yAxis;
+        const firstYAxisIndex = result[0] ? result[0].yAxis : 0;
```

Comment:
consider safer condition `result.lenght > 0 ? result[0].yAxis : 0` :)

Reply:
it's done 👍